client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -291,6 +345,68 @@ class WooPayDirectCheckout {
 		);
 	}
 
+	/**
+	 * Gets the WooPay redirect data.
+	 *
+	 * @return {Promise<Promise<*>|*>} Resolves to the WooPay redirect response.
+	 */
+	static async getWooPayMinimumSesssionDataFromMerchant() {
+		// This should always be defined, but fallback to a request in case of the unexpected.
+		if ( getConfig( 'woopayMinimumSessionData' ) ) {
+			return getConfig( 'woopayMinimumSessionData' );
+		}
+
+		return request(
+			buildAjaxURL(
+				getConfig( 'wcAjaxUrl' ),
+				'get_woopay_minimum_session_data'
+			),
+			{
+				_ajax_nonce: getConfig( 'woopaySessionNonce' ),
+			}
+		);
+	}
+
+	/**
+	 * Validates a WooPay redirect URL.
+	 *
+	 * @param {string} redirectUrl The URL to validate.
+	 * @param {string} requiredParam The URL parameter that is required in the URL.
+	 *
+	 * @return {boolean} True if URL is valid, false otherwise.
+	 */
+	static validateRedirectUrl( redirectUrl, requiredParam ) {
+		try {
+			const parsedUrl = new URL( redirectUrl );
+			if (
+				parsedUrl.origin !== getConfig( 'woopayHost' ) ||
+				! parsedUrl.searchParams.has( requiredParam )
+			) {
+				return false;
+			}
+
+			return true;
+		} catch ( error ) {
+			return false;
+		}
+	}
+
+	/**
+	 * Validate minimum session data.
+	 *
+	 * @param {Object} encryptedSessionData The encrypted session data to be used in URL.
+	 *
+	 * @return {boolean} True if all required fields are present, false otherwise.
+	 */
+	static validateEncryptedSessionData( encryptedSessionData ) {
+		return (
+			encryptedSessionData?.blog_id &&
+			encryptedSessionData?.data?.session &&
+			encryptedSessionData?.data?.iv &&
+			encryptedSessionData?.data?.hash
+		);
+	}
```

Comment:
There's a very similar method in this file – `isValidEncryptedSessionData`.

Interesting how we sometimes end up writing almost the same code in separate PRs without knowing. 😅

Reply:
😆 Wow, can't believe I missed that! Now using just the existing function and removed the duplicate in https://github.com/Automattic/woocommerce-payments/pull/8268/commits/4b1b7d4c320e178232d3c1ebea1cfed829d7c355